### PR TITLE
add errorBoundary and tests

### PR DESCRIPTION
**Pull request by Yoland172**

### CODE REVIEW

File: src/components/errorBoundary/ErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { Component, ErrorInfo, ReactNode } from "react";
+import styles from "./errorBoundary.module.scss";
+import errorImage_dark from "../../assets/img/fatalError_dark.png";
+import errorImage_light from "../../assets/img/fatalError_light.png";
+import i18next from "i18next";
+import classNames from "classnames";
+
+interface Props {
```
Comment (Anonim147) [inline src/components/errorBoundary/ErrorBoundary.tsx]: ErrorBoundaryProps

File: src/components/errorBoundary/ErrorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  children?: ReactNode,
+  theme:string
+}
+
+interface State {
```
Comment (Anonim147) [inline src/components/errorBoundary/ErrorBoundary.tsx]: ErrorBoundaryState

File: src/components/errorBoundary/ErrorBoundary.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  hasError: boolean;
+}
+
+class ErrorBoundary extends Component<Props, State> {
+  public state: State = {
```
Comment (Anonim147) [inline src/components/errorBoundary/ErrorBoundary.tsx]: remove public modifier. 
You don't need this methods outside of ErrorBoundary class

File: src/components/errorBoundary/ErrorBoundary.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    hasError: false,
+  };
+
+  public static getDerivedStateFromError(_: Error): State {
+    // Update state so the next render will show the fallback UI.
```
Comment (Anonim147) [inline src/components/errorBoundary/ErrorBoundary.tsx]: remove comment

File: src/components/errorBoundary/ErrorBoundary.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    return { hasError: true };
+  }
+   public componentDidMount(): void {
+   console.log(this.props.theme);
```
Comment (Anonim147) [inline src/components/errorBoundary/ErrorBoundary.tsx]: remove

File: src/components/errorBoundary/ErrorBoundary.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    console.log("Uncaught error:", error, errorInfo);
+  }
+
+  public render() {
+    if (this.state.hasError) {
+      return (
+        <div className={classNames(styles.errorContainer, this.props.theme && styles[this.props.theme])}>
+            <img src={this.props.theme === "light" ? errorImage_light : errorImage_dark} alt="errorImage"/>
+          <h1> {i18next.t("errorPage.wrongText")}</h1>
+          <h3>{i18next.t("errorPage.tryAgain_1")} <span onClick={() => window.location.reload()}>{i18next.t("errorPage.tryAgain_2")}</span></h3>
```
Comment (Anonim147) [inline src/components/errorBoundary/ErrorBoundary.tsx]: please format it in correct way
```
<h3>
      <span> </span>
</h3>
```

File: src/components/errorBoundary/ErrorBoundary.tsx (hunk 1 above)
Comment (Yoland172) [reply src/components/errorBoundary/ErrorBoundary.tsx]: fixed

File: src/components/errorBoundary/ErrorBoundary.tsx (hunk 2 above)
Comment (Yoland172) [reply src/components/errorBoundary/ErrorBoundary.tsx]: fixed

File: src/components/errorBoundary/ErrorBoundary.tsx (hunk 3 above)
Comment (Yoland172) [reply src/components/errorBoundary/ErrorBoundary.tsx]: fixed

File: src/components/errorBoundary/ErrorBoundary.tsx (hunk 4 above)
Comment (Yoland172) [reply src/components/errorBoundary/ErrorBoundary.tsx]: fixed

File: src/components/errorBoundary/ErrorBoundary.tsx (hunk 5 above)
Comment (Yoland172) [reply src/components/errorBoundary/ErrorBoundary.tsx]: fixed

File: src/components/errorBoundary/ErrorBoundary.tsx (hunk 6 above)
Comment (Yoland172) [reply src/components/errorBoundary/ErrorBoundary.tsx]: fixed

**Review by Anonim147 — APPROVED** [review]

**Pull request merged**